From 2fc47feff98ca5a0c1dcc192df840efd0b28027e Mon Sep 17 00:00:00 2001
From: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date: Mon, 24 Jul 2017 15:09:15 +0300
Subject: [PATCH] ADMINUI-2378 Adminui should trim leading and trailing
 whitespace from search input fields Reviewed by: Angela Fong
 <angela.fong@joyent.com> Approved by: Angela Fong <angela.fong@joyent.com>

---
 lib/adminui.js | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/lib/adminui.js b/lib/adminui.js
index 768ab268..d0c47d6e 100644
--- a/lib/adminui.js
+++ b/lib/adminui.js
@@ -32,12 +32,20 @@ var notes = require('./notes');
 var settings = require('./settings');
 
 function resume(req, res, next) {
+
     process.nextTick(function () {
         req.resume();
     });
     next();
 }
 
+function paramsTrimmer(req, res, next) {
+    Object.keys(req.params).forEach(function (paramName) {
+        req.params[paramName] = req.params[paramName].trim();
+    });
+    next();
+}
+
 function setUfds(req, res, next) {
     var trySetUfds = function () {
         if (req.ufdsMaster.connected) {
@@ -147,6 +155,7 @@ ADMINUI.prototype.createServer = function () {
     server.use(restify.queryParser());
     server.use(restify.gzipResponse());
     server.use(restify.acceptParser(server.acceptable));
+    server.use(paramsTrimmer);
 
     TraceEvent.mount({
         skipRoutes: ['ping', 'getca'],
-- 
2.21.0

